src/ApiCheck/ApiListingGenerator.cs

```diff
@@ -0,0 +1,409 @@
+                case MemberTypes.Method:
+                    var name = member.Name;
+                    var method = (MethodInfo)member;
+                    var methodDescriptor = new MemberDescriptor();
+
+                    methodDescriptor.Kind = MemberKind.Method;
+
+                    methodDescriptor.Visibility = method.IsPublic ? ApiElementVisibility.Public :
+                        method.IsFamilyOrAssembly ? ApiElementVisibility.ProtectedInternal :
+                        method.IsFamily ? ApiElementVisibility.Protected :
+                        method.IsPrivate ? ApiElementVisibility.Private : ApiElementVisibility.Internal;
+
+                    if (!type.IsInterface)
+                    {
+                        methodDescriptor.ExplicitInterface = GetInterfaceImplementation(method, explicitImplementation: true);
+                        methodDescriptor.ImplementedInterface = methodDescriptor.ExplicitInterface ?? GetInterfaceImplementation(method, explicitImplementation: false);
+                    }
+
+                    methodDescriptor.Name = MemberDescriptor.GetMemberNameFor(method);
+
+                    if (method.IsGenericMethod)
+                    {
+                        var constraints = GetGenericConstraintsFor(method.GetGenericArguments().Select(t => t.GetTypeInfo()));
+                        foreach (var constraint in constraints)
+                        {
+                            methodDescriptor.GenericConstraints.Add(constraint);
+                        }
+                    }
+
+                    methodDescriptor.Static = method.IsStatic;
+                    methodDescriptor.Sealed = method.IsFinal;
+                    methodDescriptor.Virtual = method.IsVirtual;
+                    methodDescriptor.Override = method.IsVirtual && method.GetBaseDefinition() != method;
+                    methodDescriptor.Abstract = method.IsAbstract;
+                    methodDescriptor.New = !method.IsAbstract && !method.IsVirtual && method.IsHideBySig &&
+                        method.DeclaringType.GetMember(method.Name).OfType<MethodInfo>()
+                        .Where(m => SameSignature(m, method)).Count() > 1;
+                    methodDescriptor.Extension = method.IsDefined(typeof(ExtensionAttribute), false);
+
+                    foreach (var parameter in method.GetParameters())
+                    {
+                        var parameterDescriptor = GenerateparameterDescriptor(parameter);
+                        methodDescriptor.Parameters.Add(parameterDescriptor);
+                    }
+
+                    methodDescriptor.ReturnType = TypeDescriptor.GetTypeNameFor(method.ReturnType.GetTypeInfo());
+
+                    return methodDescriptor;
+                case MemberTypes.Field:
+                    var field = (FieldInfo)member;
+                    if (type.IsEnum && !field.IsLiteral)
+                    {
+                        // Skip storage for enumerations.
```

Comment:
Hmm does this mean it's not recording the values of enums? It probably should be, because that's also a really bad breaking change, no? Or is this something else?